common/src/main/scala/org/neo4j/spark/util/Neo4jImplicits.scala

```diff
@@ -198,4 +200,31 @@ object Neo4jImplicits {
     def groupByCols(): Array[Expression] = ReflectionUtils.groupByCols(aggregation)
   }
 
+  implicit class MapImplicit[K, V](map: Map[String, V]) {
+    def flattenMap(prefix: String = ""): Map[String, AnyRef] = {
```

Comment:
I used that name because there is already a `flatten` in scala GenericTraversableTemplate:
https://www.scala-lang.org/api/2.12.7/scala/collection/Iterable.html
We can think about another like `flatToDottedKeys` (this one no, is very ugly :smile:)